enclave/core/malloc.c

```diff
@@ -150,12 +90,33 @@ int oe_posix_memalign(void** memptr, size_t alignment, size_t size)
 
 size_t oe_malloc_usable_size(void* ptr)
 {
-    if (oe_use_debug_malloc)
-    {
-        return MALLOC_USABLE_SIZE(ptr);
-    }
-    else
-    {
-        return oe_allocator_malloc_usable_size(ptr);
-    }
+    return oe_allocator_malloc_usable_size(ptr);
+}
+
+oe_result_t oe_check_memory_leaks(void)
+{
+    // Without debug malloc, no leaks are reported.
+    return OE_OK;
+}
+
+oe_result_t oe_debug_malloc_tracking_start(void)
+{
+    return OE_OK;
+}
+
+oe_result_t oe_debug_malloc_tracking_stop(void)
+{
+    return OE_OK;
+}
+
+oe_result_t oe_debug_malloc_tracking_report(
+    uint64_t* out_num_objects,
+    char** report)
+{
+    *out_num_objects = (uint64_t)-1;
```

Comment:
Why -1 instead of 0 which would seem to make more sense?

Reply:
While the enclave is not linked against liboedebugmalloc.a, the function oe_debug_malloc_tracking_report() in malloc.c will be loaded. -1 may alert users there are not using debug malloc. Same reason the result is set to OE_FAILURE on purpose.

Reply:
Now it is initialized as 0, thanks!